### ISSUE TITLE
IKEv2 Support

### ISSUE THREAD

**Issue by kronicd** [pia-foss/manual-connections#178]
I'd like to request IKEv2 support be added to the manual connections scripts.

I'm more than happy to contribute this myself but I'm having some trouble understanding how the usernames & passwords are generated for IKEv2 use.

I've had a look through the code in [client-library-apple](https://github.com/pia-foss/client-library-apple) however I am unfamiliar with the iOS VPN management APIs and it is a little hard to follow.

Any insights would be appreciated.

**Comment by kohnish** [comment]
Any new findings?  

I'm not faimilar with iOS development but judging from https://developer.apple.com/documentation/devicemanagement/vpn/ikev2-data.dictionary, AuthenticationMethod and ExtendedAuthEnabled values suggest it's EAP.  
  
But I'm at a loss with user password part as well. I'm going to try a few things with strongswan. It's an open protocol and done by apple, it should work with any normal IKEv2 client.

**Comment by kohnish** [comment]
Updated the PR to include a working example. To be polished later.
The issue was that I was using wrong ID and correct CA certs weren't imported in strongswan.
